lib/graph3d/Settings.js

```diff
@@ -493,65 +509,98 @@ function setSurfaceColor(surfaceColors, dst) {
-    }
-    if(surfaceColors.hue.colorStops < 2) {
-      throw new Error('Surface colors colorStops is out of bounds. Expected 2 or above.');
-    }
+    rgbColors = parseColorObject(surfaceColors.hue);
+  } else {
+    throw new Error('Unsupported type of surfaceColors');
+  }
+  // for some reason surfaceColors goes from vMax to vMin:
+  rgbColors.reverse();
+  dst.colormap = rgbColors;
+}
 
-    const startHue = Math.min(surfaceColors.hue.start, surfaceColors.hue.end);
-    const endHue = Math.max(surfaceColors.hue.start, surfaceColors.hue.end);
-    const saturation = surfaceColors.hue.saturation;
-    const brightness = surfaceColors.hue.brightness;
-    const stops = surfaceColors.hue.colorStops;
-    const hueStepSize = (endHue - startHue) / stops;
-
-    let currentHue = startHue;
-    while (currentHue < endHue) {
-      let calculatedHue = currentHue % 360;
-      if(calculatedHue < 0) {
-        calculatedHue = 360 + calculatedHue;
-      }
-      rgbColors.push(
-        util.HSVToRGB(
-          calculatedHue/360,
-          saturation/100, 
-          brightness/100
-        )
-      );
-      currentHue += hueStepSize;
-    }
 
-    if (surfaceColors.hue.start > surfaceColors.hue.end) {
-      rgbColors.reverse();
-    }
+/**
+ *
+ * @param {Object | Array<string>} colormap Either an object that describes the HUE, or an array of HTML hex color codes
+ * @param {Object} dst
+ */
+function setColormap(colormap, dst) {
+  if (colormap === undefined) {
+    return;
+  }
+
+  let rgbColors;
+  if (Array.isArray(colormap)) {
+    rgbColors = parseColorArray(colormap);
+  } else if (typeof colormap === 'object') {
+    rgbColors = parseColorObject(colormap.hue);
+  } else if (typeof colormap === 'function') {
+    rgbColors = colormap;
   } else {
-    throw new Error('Unsupported type of surfaceColors');
+    throw new Error('Unsupported type of colormap');
   }
+  dst.colormap = rgbColors;
+}
 
-  dst.surfaceColors = rgbColors;
+
+/**
+ *
+ * @param {Array} colormap
+ */
+function parseColorArray(colormap) {
+  if(colormap.length < 2) {
+    throw new Error('Colormap array length must be 2 or above.');
+  }
+  return colormap.map(function(colorCode){
+    if(!util.isValidHex(colorCode)) {
+      throw new Error(`Invalid hex color code supplied to colormap.`);
+    }
+    return util.hexToRGB(colorCode);
+  });
 }
 
+
+/**
+ * Converts an object to a certain amount of hex color stops. At which point:
+ * the HTML hex color codes is converted into an RGB color object.
+ *
+ * @param {Object} hues
+ */
+function parseColorObject(hues) {
+  if (hues === undefined) {
+    throw new Error('Unsupported type of colormap');
+  }
+  if (hues.saturation < 0 || hues.saturation > 100) {
+    throw new Error('Saturation is out of bounds. Expected range is 0-100.');
+  }
+  if (hues.brightness < 0 || hues.brightness > 100) {
+    throw new Error('Brightness is out of bounds. Expected range is 0-100.');
+  }
```

Comment:
Generally speaking don't test that something is wrong, test that something is okay (you can always wrap it in `!(…)`). In this case NaN isn't smaller than 0, neither it's higher than 100, but that doesn't mean it's valid.

Reply:
Ok, I just moved this code without change but I can do this style fix.